### PR TITLE
chore(E2E) : E2E tests for status grid widget

### PR DESCRIPTION
**Pull request by MedWassCh**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [ ] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+    command: `su -s /bin/sh ${apacheUser} -c "/usr/bin/env php -q /usr/share/centreon/cron/centAcl.php"`,
+    name: Cypress.env('dockerName')
+  });
+});
+
+beforeEach(() => {
+  cy.intercept({
+    method: 'GET',
+    url: '/centreon/api/internal.php?object=centreon_topology&action=navigationList'
+  }).as('getNavigationList');
```
[the remark this answers is not in the document]
Comment (MedWassCh) [reply centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts:157]: I'm using the get navigation list in before when i login as an admin and also as in beforeEach scenarios also

[the remark this answers is not in the document]
Comment (kduret) [reply centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts:157]: then a single declaration should work, isn't it ?

File: centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    cy.get('[data-status="critical"]').should('exist');
+  }
+);
+
+Given('a dashboard featuring two Status Grid widgets', () => {
+  cy.insertDashboardWithStatusGridWidget(
+    dashboards.default,
+    twoStatusGridWidgets
+  );
+  cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
```
Comment (kduret) [inline centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts:288]:
```suggestion
  cy.visit('/centreon/home/dashboards');
```

File: centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MedWassCh) [reply centreon/tests/e2e/features/Dashboards/12-status-grid-widget-configuration/index.ts:156]: In fact in loginByTypeOfUser we need to wait for getNavigationList and if i remove it it doesn't work

**Review by kduret — APPROVED** [review]

**Review by paul-oureib — APPROVED** [review]

**Pull request merged**